src/utils/makeReactNativeConfig.js

```diff
@@ -226,14 +223,45 @@ function makeReactNativeConfig(
         : userWebpackConfig
     );
 
-    // For simplicity, we don't require users to extend
-    // default config.entry but do it for them.
-    config.entry = defaultWebpackConfig.entry.concat(config.entry);
+    config.entry = injectPolyfillIntoEntry(config.entry, polyfillPath);
 
     return config;
   });
 
   return [configs, PLATFORMS];
 }
 
+/*
+ * Takes user entries for webpack.haul.js,
+ * change them to multi-point entries
+ * and injects polyfills
+ */
+function injectPolyfillIntoEntry(
+  userEntry: WebpackEntry,
+  polyfillPath: string
+): WebpackEntry {
+  if (typeof userEntry === 'string') {
+    return [polyfillPath, userEntry];
+  }
+  if (Array.isArray(userEntry)) {
+    return [].concat(polyfillPath, ...userEntry);
+  }
+  if (typeof userEntry === 'object') {
+    const chunkNames = Object.keys(userEntry);
+    return chunkNames.reduce((entryObj, name) => {
+      const chunk = userEntry[name];
+      if (typeof chunk === 'string') {
+        entryObj[name] = [polyfillPath, chunk];
+        return entryObj;
+      } else if (Array.isArray(chunk)) {
+        entryObj[name] = [].concat(polyfillPath, ...chunk);
+        return entryObj;
+      }
```

Comment:
This is duplicated logic from above, maybe we could reuse it?

Reply:
Recursion would be nice